_plugins/picture.rb

```diff
@@ -0,0 +1,22 @@
+module Jekyll
+class RenderTimeTag < Liquid::Tag
+
+    def initialize(tag_name, markup, tokens)
+        super
+        @args = Parser.new(markup)
+    end
+
+    def render(context)
+        width = "width=#{@args[:width]}"
+        tag = "<picture>
+        <source srcset='./assets/images/#{@args[:src]}.webp' type='image/webp'>
+        <img src='./assets/images/#{@args[:src]}.png' alt='#{@args[:alt] unless @args[:alt].nil?}' #{width unless @args[:width].nil?}>
+        </picture>
```

Comment:
Rather than checking for each attribute like this, can we just loop over all attributes and add them here?

Reply:
@astronomersiva Can you check now ?

Reply:
this looks good...I believe the only issue now is with versioning the files(the checksum)?

Reply:
@astronomersiva Yes. Do you have any suggestions wrt versioning ?

Reply:
Can you try using the jekyll assets helper here? Let's try that out and see